conans/model/recipe_ref.py

```diff
@@ -109,3 +110,23 @@ def loads(rref):  # TODO: change this default to validate only on end points
             raise ConanException(
                 f"{rref} is not a valid recipe reference, provide a reference"
                 f" in the form name/version[@user/channel]")
+
+    def matches(self, pattern, is_consumer):
+        negate = False
+        if pattern.startswith("!"):
+            pattern = pattern[1:]
+            negate = True
+
+        condition = ((pattern == "&" and is_consumer) or
+                      fnmatch.fnmatchcase(str(self), pattern) or
+                      fnmatch.fnmatchcase(self.repr_notime(), pattern))
+        if negate:
+            return not condition
+        return condition
+
+
+def ref_matches(ref, pattern, is_consumer):
+    if not ref or not str(ref):
+        assert is_consumer
+        ref = RecipeReference.loads("*/*")
```

Comment:
Not sure about this ``RecipeReference.loads()`` with something that will fail if we start to add some validation there (it was removed in the refactor, but at some point it might make sense to do some checks). If it is going to match always, can't we return True directly?

Reply:
I agree that is ugly. But no, you cannot return True directly, maybe the pattern is `!&` or an invalid one like `potato`.